modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/ChangeAPIBackendURLTestCase.java

```diff
@@ -0,0 +1,132 @@
+        createPublishAndSubscribeAPI(apiIdentifierAPI1Version1, API1_CONTEXT, apiPublisherClientUser1,
+                apiStoreClientUser1, applicationName);
+
+        //get access token
+        String accessToken = getAccessToken(apiStoreClientUser1, applicationName);
+
+        // Create requestHeaders
+        requestHeaders = new HashMap<String, String>();
+        requestHeaders.put("Authorization", "Bearer " + accessToken);
+
+        //Invoke  old version
+        HttpResponse oldVersionInvokeResponse =
+                HttpRequestUtil.doGet(API_BASE_URL + API1_CONTEXT + "/" + API_VERSION1 + API1_END_POINT_METHOD,
+                        requestHeaders);
+        assertEquals(oldVersionInvokeResponse.getResponseCode(), HTTP_RESPONSE_CODE_OK,
+                "Response code mismatched when invoke api before change the backend URL");
+        assertTrue(oldVersionInvokeResponse.getData().contains(API1_RESPONSE_DATA),
+                "Response data mismatched when invoke  API  before change the backend URL" +
+                        " Response Data:" + oldVersionInvokeResponse.getData());
+
+    }
+
+
+    @Test(groups = {"wso2.am"}, description = "Test changing of the API Backend URL",
+            dependsOnMethods = "testAPIInvocationBeforeChangeTheBackendURL")
+    public void testEditBackendURLAPIContext() throws Exception {
+
+
+        //Create the API Request with new context
+
+        APIRequest apiRequestBean = new APIRequest(API1_NAME, API1_CONTEXT, new URL(API2_END_POINT_URL));
+        apiRequestBean.setTags(API1_TAGS);
+        apiRequestBean.setDescription(API1_DESCRIPTION);
+        apiRequestBean.setVersion(API_VERSION1);
+        apiRequestBean.setVisibility("public");
+        //Update API with Edited information
+        HttpResponse updateAPIHTTPResponse = apiPublisherClientUser1.updateAPI(apiRequestBean);
+
+        assertEquals(updateAPIHTTPResponse.getResponseCode(), HTTP_RESPONSE_CODE_OK,
+                "Update API backend URL Response Code is invalid." + getAPIIdentifierString(apiIdentifierAPI1Version1));
+        assertEquals(getValueFromJSON(updateAPIHTTPResponse, "error"), "false",
+                "Error in API backend URL Update in " + getAPIIdentifierString(apiIdentifierAPI1Version1) +
+                        "Response Data:" + updateAPIHTTPResponse.getData());
+
+    }
+
+
+    @Test(groups = {"wso2.am"}, description = "Test the invocation of API using new backend URL" +
+            "  after backend URL  change", dependsOnMethods = "testEditBackendURLAPIContext")
+    public void testInvokeAPIAfterChangeAPIBackendURLWithNewBackendURL() throws Exception {
+        //Invoke  new context
+        HttpResponse oldVersionInvokeResponse =
+                HttpRequestUtil.doGet(API_BASE_URL + API1_CONTEXT + "/" + API_VERSION1,
+                        requestHeaders);
+        assertEquals(oldVersionInvokeResponse.getResponseCode(), HTTP_RESPONSE_CODE_OK,
```

Comment:
add details about expected response code to error message, so users can understand what went wrong exactly.